pkg/fab/discovery/discovery.go

```diff
@@ -29,20 +28,27 @@ const (
 	signerCacheSize = 10 // TODO: set an appropriate value (and perhaps make configurable)
 )
 
+//Client gives ability to send discovery request to multiple targets.
+//There are cases when multiple targets requested and some of them are hanging, recommended to cancel ctx after first successful response.
+//Note: "access denied" is a success response, so check for it after response evaluation.
+type Client interface {
+	Send(ctx context.Context, req *Request, targets ...fab.PeerConfig) (<-chan Response, error)
+}
+
 // Client implements a Discovery client
-type Client struct {
+type client struct {
 	ctx      fabcontext.Client
 	authInfo *discovery.AuthInfo
 }
 
 // New returns a new Discover client
-func New(ctx fabcontext.Client) (*Client, error) {
+func New(ctx fabcontext.Client) (Client, error) {
```

Comment:
This function should return a pointer to a struct and not an interface (according to Go conventions).

Reply:
I was following https://golang.org/doc/effective_go.html#generality. 

Good, even I'm not a fun of `accept interfaces, return structs` :)

Reply:
> I was following https://golang.org/doc/effective_go.html#generality.
> 
> Good, even I'm not a fun of `accept interfaces, return structs` :)

Hmm - I guess I learned something new today :)